GUI/Util/SearchModifications.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Windows.Controls;
+using System.Windows.Threading;
+
+namespace MetaMorpheusGUI.Util
+{
+    class SearchModifications
+    {
+        public DispatcherTimer Timer;
+        public bool TimerCreated { get; set; }
+        public bool FixedSearch { get; set; }
+        public bool VarSearch { get; set; }
+
+        public SearchModifications()
+        {
+            Timer = new DispatcherTimer();
+            Timer.Interval = TimeSpan.FromMilliseconds(300);
+            TimerCreated = false;
+        }
+
+        public void SetTimer()
+        {
+            Timer.Stop(); // Resets the timer
+            Timer.Start();
+        }
+
+        public void FilterTree(TextBox textbox, TreeView tree, ObservableCollection<ModTypeForTreeView> collection)
+        {
```

Comment:
the major feature I'd like to see here is I want the mod type visible along with the expanded node and the modifications that meet the search criteria. e.g.:

Currently:
```
[] Mod1
[] Mod2
[] Mod3
```

I'd like:
```
[] ModType1
   [] Mod1
   [] Mod2
[] ModType2
   [] Mod3
```